### DISCUSSION TITLE
how to start write some code?

### DISCUSSION THREAD

**Discussion by Richard-happy** [libMesh/libmesh#3723] [Q&A]
I have completed the step **make install** . Now I want to try to write codes. Please tell me if I want to write C code ,which file path I should put the file in and what commands I should run. Please help me

**Comment by roystgnr** [comment]
You want to write C++ code (although we use the suffix .C for those, somewhat confusingly); almost none of libMesh has pure-C shims.  You can put your new files in any path you choose.  There are a dozen different ways to set up your build system, but if you're looking for "just *any* way" as a starting point, I'd suggest going to your install directory, then `cd introduction/ex1/ && make` - our example programs get installed alongside the library, and those example programs each have a Makefile which is a decent starting point for a small project.  Copy an example along with its accompanying Makefile to a new directory and modify from there.
